src/akvo/flow_services/uploader.clj

```diff
@@ -112,10 +117,56 @@
                              :port 443}]
       (gae/put! ds "Message" msg))))
 
+(defn- retrieve-questions [bucket-name surveyId]
+  (let [settings @config/settings
+        config (config/find-config bucket-name)]
+    (gae/with-datastore [ds {:server (:domain config)
+                             :email (:username settings)
+                             :password (:password settings)
+                             :port 443}]
+      (let [query (Query. "Question")
+          qf (.setKeysOnly (.setFilter query (gae/get-filter "surveyId" (Long/valueOf surveyId))))
+          pq (.prepare ds qf)]
+        (try
+          (.asList pq (gae/get-fetch-options))
+          (catch Exception e
+            (errorf e "Error retrieving questions for survey %s from GAE: %s" surveyId (.getMessage e))))))))
+
+(defn- get-datastore-ids [bucket-name surveyId]
+  (for [question (retrieve-questions bucket-name surveyId)]
+    (.getId (.getKey question))))
+
+(defn- get-file-ids [sheet]
+  (map (fn [idstr] (Long. idstr))
+       (filter (fn [idstr] (not (nil? idstr)))
+                 (map (fn [cell] (nth (re-find #"(\d+)|" (.getStringCellValue cell)) 1))
+                      (.getRow sheet 0)))))
```

Comment:
So this is kind of a "style" issue but when working with sequence functions it's often quite nice to use the [`->>`](http://clojure.github.io/clojure/clojure.core-api.html#clojure.core/->>) macro which removes most of the nesting and makes the sequence of steps more explicit:

```clojure
(->> (.getRow sheet 0)
     (map (fn [cell] (nth (re-find #"(\d+)|" (.getStringCellValue cell)) 1)))
     (filter (fn [idstr] (not (nil? idstr))))
     (map (fn [idstr] (Long. idstr))))
```

Some other things that we could change in this expression:
* Use `second` instead of `(nth __ 1)`
* Use [`some?`](http://clojure.github.io/clojure/clojure.core-api.html#clojure.core/some?) instead of `(not (nil? ___))`
* Use `(Long/parseLong __)` instead of `(Long. __)`
* For small anonymous functions it can also be more succinct and readable to use the function shorthand `#(...)` instead of `(fn [x] ...)`. 

With these comments in mind the expression would look something like 

```clojure
(->> (.getRow sheet 0)
     (map (fn [cell] (second (re-find #"(\d+)|" (.getStringCellValue cell)))))
     (filter some?)
     (map #(Long/parseLong %))
```

This reads quite nicely top-to-bottom: 
* get the first row, 
* map a regex parsing function over the columns, 
* drop nil values, 
* parse to Long

(Note that Java static methods (`Long/parseLong`) are not functions (they're methods!) so you need to wrap it in a function like #(Long/parseLong %) (which is equivalent to `(fn [s] (Long/parseLong s))`).

Does this function run through all the columns for row 0? If that is the case (and if the first `nil` column marks the end of the data report columns) we could possibly use [`(take-while some?)`](http://clojure.github.io/clojure/clojure.core-api.html#clojure.core/take-while) instead of `(filter some?)`. That way we would would stop the parsing when we reach the end of the report.